src/services/package-manager/package-service.ts

```diff
@@ -78,23 +112,242 @@ class PackageService {
+        await this.publishPackage(packageToImport);
+        logger.info(`Imported package with key: ${packageToImport.packageKey} successfully`)
+    }
+
+    private async updateDependencyVersions(node: ManifestNodeTransport): Promise<void> {
+        const createdNode = await nodeApi.findOneByKeyAndRootNodeKey(node.packageKey, node.packageKey);
+        for (const dependency of node.dependencies) {
+            const nodeInTargetTeam = await nodeApi.findOneByKeyAndRootNodeKey(dependency.key, dependency.key);
+            const nextVersion = await packageApi.findActiveVersionById(nodeInTargetTeam.id);
+            dependency.version = nextVersion.version;
+            dependency.updateAvailable = false;
+            dependency.id = nodeInTargetTeam.rootNodeId;
+            dependency.rootNodeId = nodeInTargetTeam.rootNodeId;
+            await packageDependenciesApi.updatePackageDependency(createdNode.id, dependency);
+        }
+    }
+
+    private async fillNodeDependencies(nodesListToExport: BatchExportNodeTransport[], allPackages: ContentNodeTransport[], actionFlowPackageKeys: string[], nodePath: string[]): Promise<BatchExportNodeTransport[]> {
+        let nodesListWithActiveVersion = await this.getNodesWithActiveVersion(nodesListToExport);
+
+        const draftIdByNodeId = new Map<string, string>();
+        const nodeIds = nodesListWithActiveVersion.map(node => {
+            draftIdByNodeId.set(node.id, node.workingDraftId);
+            return node.id;
+        });
+
+        const dependenciesByPackageId = await this.getPackagesDependenciesByPackageId(nodeIds, draftIdByNodeId, actionFlowPackageKeys);
+
+        nodesListWithActiveVersion = nodesListWithActiveVersion.map(nodeToExport => {
+            nodeToExport.dependencies = dependenciesByPackageId.get(nodeToExport.id) ?? [];
+            return nodeToExport;
+        })
+
+        const variableAssignments = await variableService.getVariableAssignmentsForNodes(nodesListWithActiveVersion);
+
+        nodesListWithActiveVersion.forEach(node => {
+            node.variables = variableAssignments.find(nodeWithVariablesAssignment => nodeWithVariablesAssignment.key === node.key)?.variableAssignments;
+        });
+
+        const dataModelAssignments = await dataModelService.getDatamodelsForNodes(nodesListToExport);
+        nodesListWithActiveVersion.forEach(node => {
+            node.datamodels = dataModelAssignments.get(node.key);
+        })
+
+        nodesListToExport = await this.getNodeDependencies(nodesListToExport, allPackages, actionFlowPackageKeys, nodePath);
+        return nodesListToExport
+    }
+
+    private async getNodeDependencies(nodesListToExport: BatchExportNodeTransport[], allPackages: ContentNodeTransport[], actionFlowPackageKeys: string[], nodePath: string[]): Promise<BatchExportNodeTransport[]> {
+        for (const node of nodesListToExport) {
+            const nodesToGetKeys = node.dependencies.filter(dependency => !nodesListToExport
```

Comment:
Nit: maybe unexportedNodeKeys would fit this better. This name doesn't seem to fit in my opinion. What do you think?